### PR TITLE
Improve lispy-comment behaviour at eol-like positions

### PR DESCRIPTION
**Pull request by sooheon**
1. When applicable, first add a single `;` comment on the same line.
Calling `lispy-comment` again turns it into a `;;` comment and
reindents.

2. For example, at eol, rather than being eager to newline and add a
`;;` comment, forward to comment-dwim first. This also applies to
eol-like points, such as lispy-right or within nested closing parens. At
any of these points, to get old behavior, you can `C-j ;`, as at
`lispy-bolp` it adds a `;;` comment.

3. For commenting out sexps behavior is unchanged. For adding comments
at eol or end-of-sexp, it behaves as if you cleared out preceding
closing parens and then called comment-dwim.

### CODE REVIEW

**Comment by sooheon** [comment]
Ah forgot to document another change--when commenting from lispy-left, i.e. commenting out sexps, commenting out the last sexp in enclosing list calls `(lispy--out-backward 1)`. IMO this is disorienting when this takes you up more than a screen, and it's not even clear what you just commented out. I think it's a case of a little *too* much dwim.

**Comment by abo-abo** [comment]
Sorry, but I disagree. By popular convention, only `;;` comments are acceptable for Elisp, which should start from an empty line. I don't know about other dialects, but I'm pretty sure `;;` is acceptable there as well.

In any case, some pointers for a new PR:

1. Make sure that `make test` succeeds after your modification (it doesn't right now).
2. Make sure to add new tests so that `make test` fails before your change is made.
3. In case of a disputed convention, like right now, add a `defcustom` that toggles the new convention. And make sure that it's off by default.

**Comment by sooheon** [comment]
Ok thanks for the feedback, will try and apply them incrementally.

**Comment by sooheon** [comment]
FWIW, I found [these](https://www.gnu.org/software/emacs/manual/html_node/elisp/Comment-Tips.html) recommendations on commenting. Still, I agree a defcustom to toggle it would be best.

**Comment by sooheon** [comment]
I'd appreciate your input. I've also changed the behaviour a bit so that a trailing space is always added (previously, at `(and (eolp) (not (lispy-rightp)))`, it just added ` ;|`. This improves muscle memory, as now, any invocation of `lispy-comment` should leave you with n semicolons and then a single space always.

File: lispy-test.el [hunk 1]
```diff
@@ -1481,7 +1481,21 @@ Insert KEY if there's no command."
   (should (string= (lispy-with "(list\n foo\n |1 (2\n    3)\n bar)" ";")
                    "(list\n foo\n |;; 1 (2\n ;;    3)\n bar)"))
   (should (string= (lispy-with "(defun foo ()|\n  (bar))" ";")
-                   "(defun foo ()\n  ;; |\n  (bar))")))
+                   "(defun foo ()\n  ;; |\n  (bar))"))
+  ;; With single semicolons
+  (setq lispy-comment-use-single-semicolon t)
```
Comment (abo-abo) [inline lispy-test.el]: This should be a `let` instead of `setq`, so that the setting applies only to these statements, and not to all the following tests.

**Comment by abo-abo** [comment]
With your changes, the test `lispy-paredit-semicolon` fails even if `lispy-comment-use-single-semicolon` isn't set. Could you fix that?

**Comment by abo-abo** [comment]
Thanks, please looks at the change again: I've rebased some code, but your tests remain intact and are passing.

**Pull request closed without merging**